content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,70 @@
++++
+path = "inside-rust/9999/12/31/redesigning-stage-0-std"
+title = "Redesigning stage 0 std"
+authors = ["Jieyou Xu"]
+
+[extra]
+team = "Bootstrap"
+team_url = "https://www.rust-lang.org/governance/teams/infra#team-bootstrap"
++++
+
+# Redesigning stage 0 std
+
+## Summary
+
+We are reworking how the stage 0 bootstrap sequence works (the sequence used to build a stage 1 compiler).
+
+- Before: a stage 0 (beta by default) compiler is used to build the in-tree std, which in turn is used to build the stage 1 compiler.
```

Comment:
I also think that having to clarify that stage0 is "technically not always the beta compiler" makes things a bit harder to understand :thinking: Almost every contributor is going to have stage0 being the beta compiler so even if its not necessarily true I do wonder if it would make sense to lie a bit with a footnote the first time and then just ignore it from then on.
```suggestion
- Before: the beta compiler[^1] is used to build the in-tree std, which in turn is used to build the stage 1 compiler.

[^1]: In practice, rarely, the stage0 compiler can be something other than the beta compiler. This doesn't matter too much for the purpose of this change though, so we've taken the liberty of conflating the two to make it easier to explain the change to bootstrapping.
```

Reply:
I used a footnote to "uhh acktually" the first instance of assuming stage 0 compiler in this blog post as an simplifying assumption, but also noted that it's not universally true.